wait/host.go

```diff
@@ -587,24 +587,32 @@ func (a *HostAwaitility) WaitForTemplateUpdateRequests(namespace string, count i
 type NotificationWaitCriterion func(a *HostAwaitility, mur *toolchainv1alpha1.Notification) bool
 
 // WaitForNotification waits until there is a Notification available with the given name and the optional conditions
-func (a *HostAwaitility) WaitForNotification(name string, criteria ...NotificationWaitCriterion) (*toolchainv1alpha1.Notification, error) {
-	var notification *toolchainv1alpha1.Notification
+func (a *HostAwaitility) WaitForNotification(name string, criteria ...NotificationWaitCriterion) ([]*toolchainv1alpha1.Notification, error) {
+	var notification []*toolchainv1alpha1.Notification
 	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
-		obj := &toolchainv1alpha1.Notification{}
-		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, obj); err != nil {
+		obj := &toolchainv1alpha1.NotificationList{}
+		if err := a.Client.List(context.TODO(), obj); err != nil {
 			if errors.IsNotFound(err) {
 				a.T.Logf("waiting for availability of notification '%s'", name)
 				return false, nil
 			}
 			return false, err
 		}
-		for _, match := range criteria {
-			if !match(a, obj) {
-				return false, nil
+
+		for _, n := range obj.Items {
```

Comment:
with a label, we would not need such an iteration loop

Reply:
Much cleaner! Updated it as part of this commit: 155dc96be86fca8f5b90c428ae30601a1c0cf8a4